railties/lib/rails/application/bootstrap.rb

```diff
@@ -56,6 +56,10 @@ module Bootstrap
         end
         Rails.logger.level = ActiveSupport::Logger.const_get(config.log_level.to_s.upcase)
 
+        broadcast_logger = ActiveSupport::BroadcastLogger.new(Rails.logger)
```

Comment:
Is it possible that when users already use `ActiveSupport::BroadcastLogger` in configs, and then they'd get an 2-level nested `BroadcastLogger` from this?

Reply:
👋 That's indeed possible, and I remember seeing this in applications frequently. I made sure that it is supported https://github.com/rails/rails/blob/a5f113f4332b3688d4431a97fa517d1b9517ceb0/activesupport/test/broadcast_logger_test.rb#L163-L175

Reply:
Sorry I wasn't clear. I didn't mean that the propagation wouldn't work. I meant the logger would be nested like this:

```
#<ActiveSupport::BroadcastLogger:0x0000000114b374b8
 @broadcasts=
  [#<ActiveSupport::BroadcastLogger:0x000000011367d9a0
    @broadcasts=
     [#<ActiveSupport::BroadcastLogger:0x00000001139b1770
       @broadcasts=
        [#<Logger:0x0000000113d76180 @default_formatter=#<Logger::Formatter:0x00000001139b1810 @datetime_format=nil>, @formatter=nil, @level=0, @logdev=nil, @progname=nil>,
         #<Logger:0x0000000113d76130 @default_formatter=#<Logger::Formatter:0x00000001139b17c0 @datetime_format=nil>, @formatter=nil, @level=0, @logdev=nil, @progname=nil>],
       @formatter=nil,
       @progname="Broadcast">],
    @formatter=nil,
    @progname="Broadcast">],
 @formatter=nil,
 @progname="Broadcast">
```

If the user manually configs broadcast logger like:

```rb
config.logger = ActiveSupport::BroadcastLogger.new(
  ::Logger.new(nil),
  ::Logger.new(nil)
)
```

I personally would be surprised if I see another single `ActiveSupport::BroadcastLogger` when calling `Rails.logger.broadcasts` in this case.

Reply:
Ah sorry I misunderstood. Yeah good point, that feels wrong. I'll open a fix

Reply:
I think related to this, I'm noticing an issue where the initially configured logger is being put into the `BroadcastLogger`, but attached `ActionController::LogSubscriber` methods are still logging to the original logger. For example:

```ruby
broadcast_logger = ActiveSupport::BroadcastLogger.new(
  config.logger # <-- Logs via LogSubscriber are being logged to this logger and not others
)
```

I haven't been able to write a test that reproduces the behavior, so it could be a strange configuration in my own codebase. I thought it was worth calling out there anyway.

Reply:
This issue is what this PR fixes, so maybe you are using a version without this fix?

Reply:
You can also check if other loggers (e.g. `ActiveRecord::Base.logger`) have the same issue. If it's only `ActionController::Base.logger` it's maybe because your app or a library modifies that variable manually.

To help debug you could create 2 initializers and see the value of `ActionController::Base.logger`

```ruby
# config/application.rb

initializer "debug_logger_before", before: "action_controller.set_configs" do
  puts ActionController::Base.logger # Should normally be nil
end

initializer "debug_logger_after", after: "action_controller.set_configs" do
  puts ActionController::Base.logger # Should be a `BroadcastLogger` instance.

  # If it's a `BroadcastLogger` instance, then something in your app or a gem
  # modifies it afterward.
  
  # If it's not a `BroadcastLogger`, it could be because the logger is overriden. 
  # Check if you have a line of code that looks like `config.action_controller.logger = ...`
end
```

Reply:
This was very helpful, thank you! To confirm, it was an issue on our end, but not the way I thought it was.

We had a test that was stubbing `add` on ActionController::Base.logger. The trick is that `add` was never being sent to that logger, only `info`. This worked previously because internally the Ruby Logger calls `add` in `info` (and other such methods). But that `add` was never being called on the `BroadcastLogger`.

We updated our test to stub the correct method and it's working. Thanks again for this debugging help!